### ISSUE TITLE
Devise support

### ISSUE THREAD

**Issue by waleedarshad** [nsarno/knock#168]

**Comment by BigChief45** [comment]
There isn't really any point on using Devise if you are going to use the Knock gem.

**Comment by waleedarshad** [comment]
Who is going to handle all logic regarding user confirmation, resent password token, forget password, change password.... these all logic we need to write ourself which does not seems obvious solution.

Would you guide me?

**Comment by zangab** [comment]
there are plenty of tutorials out there which focus on confirmation / password-reset / etc ... google it

**Comment by vikonava** [comment]
I agree it would be way better if there is devise support for this gem. Why reinvent the wheel? thats the whole purpose of using gems. So I don't think making the confirmation/reset/etc manually is a valid answer for this request. It is not necessary but it would be a nice feature to have.

**Comment by durfee** [comment]
It doesn't look like it's something the contributors are going to do (see: https://github.com/nsarno/knock/issues/140#issuecomment-279282913). If you're looking for a guide, [Michael Hartl's Rails Tutorial](https://www.railstutorial.org/book) goes over this stuff pretty extensively (passwords, confirmations, bcrypt, etc). After you get the hang of it, it's pretty painless to set up.

**Comment by BKSpurgeon** [comment]
I have an existing rails app. Now the time has come to add JSON API features to this app.

I am currently using devise.

Do you have any ideas on how I can manage this using rails 5?

advise much appreciated.

chrs
